client/shared/src/search/query/diagnostics.ts

```diff
@@ -43,23 +60,125 @@ export function checkFilter(filter: Filter): Monaco.editor.IMarkerData[] {
     const checks: FilterCheck[] = [validFilterValue, emptyFilterValue]
     return checks.map(check => check(filter)).find(value => value.length !== 0) || []
 }
-const rules: PatternOf<Token[], Monaco.editor.IMarkerData[]>[] = [
+
+// There should be a better way to pass additional data to patterns, or for
+// patterns to keep internal state than piggybacking on `data`
+// (tracked in https://github.com/sourcegraph/sourcegraph/issues/25070)
```

Comment:
```suggestion
// TODO(fkling): Improve how we pass additional data to patterns #25070
```

nit: would make this fit a single line, since the issue has the details